test/components/all_components.rb

```diff
@@ -0,0 +1,18 @@
+require 'component_test_helper'
+
+# TODO: Move this logic to the GOVUK Publishing Components gem
+# and have the tests visit each component.
+class AllComponentsTest < ActionDispatch::IntegrationTest
+  test "renders all component guide pages without erroring" do
+    visit '/component-guide'
+    components = all(:css, '.component-list a').map do |el|
+      {
+        text: el[:text],
+        href: el[:href]
+      }
+    end
+    components.each do |component|
+      visit component[:href] + '/preview'
+    end
```

Comment:
That's better, I think I was tired when I wrote this because there's also unneeded data in there too. Will update